### PR TITLE
CP2K: Fix multiple use of spec["fftw"]

### PR DESCRIPTION
**Pull request by amd-toolchain-support**
FFTW recipe internally uses a reference to `self.spec.last_query`.  In the CP2K recipe an `fftw` object is originally created with `spec["fftw:openmp"]` if OpenMP is required, but a later fixed reference to `spec["fftw"]` overwrites the 'last_query' in the spec object, so later use of fftw.libs was not returing FFTW OpenMP libs.

Also allow the post-install package config fixup to support amdfftw as well as fftw.

### CODE REVIEW

File: var/spack/repos/builtin/packages/cp2k/package.py [hunk 1]
```diff
@@ -888,7 +888,9 @@ def fix_package_config(self):
             content += " " + self.spec["lapack"].libs.ld_flags
             content += " " + self.spec["fftw-api"].libs.ld_flags
 
-            if (self.spec["fftw-api"].name == "fftw") and ("+openmp" in self.spec["fftw"]):
+            if (self.spec["fftw-api"].name in ["fftw", "amdfftw"]) and self.spec[
+                "fftw-api"
+            ].satisfies("+openmp"):
```
Comment (becker33) [inline var/spack/repos/builtin/packages/cp2k/package.py]:
```suggestion
            fftw = self.spec["fftw-api"]
            if fftw.name in ["fftw", "amdfftw"] and fftw.satisfies("+openmp"):
```
Not a big deal, but this might be easier to read.

**Review by becker33 — DISMISSED** [review]
LGTM, optional suggestion for clarity if you agree.

**Review by mtaillefumier — APPROVED** [review]
that's alright on my side as well. You may consider using the cmake version of the build system unless there is a strong constraint on the version of CP2K.

File: var/spack/repos/builtin/packages/cp2k/package.py (hunk 1 above)
Comment (amd-toolchain-support) [reply var/spack/repos/builtin/packages/cp2k/package.py]: That makes sense to me

File: var/spack/repos/builtin/packages/cp2k/package.py [hunk 2]
```diff
@@ -888,7 +888,8 @@ def fix_package_config(self):
             content += " " + self.spec["lapack"].libs.ld_flags
             content += " " + self.spec["fftw-api"].libs.ld_flags
 
-            if (self.spec["fftw-api"].name == "fftw") and ("+openmp" in self.spec["fftw"]):
+            fftw = self.spec["fftw-api"]
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/cp2k/package.py:891]: Why is this needed, given the definition at line 367 ?

Comment (amd-toolchain-support) [reply var/spack/repos/builtin/packages/cp2k/package.py:891]: Is Spack doing something special to the Python locals dict? Otherwise I don't believe the definition at line 367 persists into the post-build fixup function.

Comment (alalazo) [reply var/spack/repos/builtin/packages/cp2k/package.py:891]: Oh, lol, my bad! From the web-gui I didn't notice Github was squeezing 500 lines in between :laughing:

**Review by alalazo — APPROVED** [review]
